### PR TITLE
Background blur on repository select open

### PR DESCRIPTION
**Pull request by Joozty**

### CODE REVIEW

File: components/application/index.jsx [hunk 1]
```diff
@@ -11,9 +10,7 @@ const Application = ({ children, dataProvider, activity, ...restProps }) => (
   <>
     <Head />
     <Container {...restProps}>
-      <AppBar>
-        <RepositorySelect value={dataProvider} />
-      </AppBar>
+      <AppBar dataProvider={dataProvider} />
```
Comment (viktor-yakubiv) [inline components/application/index.jsx]: Why could not you keep this as it was?

**Comment by viktor-yakubiv** [comment]
I approve 9446b28 👍. Feel free to cherry pick it directly to `master` and then rebase this PR. Thank you!

---

Do you have any ideas how to add global overlay? For modals and drop-downs.

File: components/application/index.jsx (hunk 1 above)
Comment (Joozty) [reply components/application/index.jsx]: I needed component connected to `LayoutContext`. (I didn't want to connect another component into it.)

**Comment by Joozty** [comment]
I don't know how to do a global overlay for dropdowns. I am worried it has to be done separately for every dropdown as it is done in this PR.


For global overlay for modals, there are several options. Nonetheless, the best option for us is to use `React.Portals`. Modal would be rendered as a direct child of body element through the portal...

```
<body>
	<div id="__next" />
	<div classname="overlay">
		<div>Modal content here</div>
	</div>
</body>
````

File: components/layout/container.jsx [hunk 2]
```diff
@@ -8,6 +8,8 @@ const reducer = (state, action) => {
   switch (action.type) {
     case 'toggle_sidebar':
       return { ...state, sidebarOpen: !state.sidebarOpen }
+    case 'toggle_repository_select':
+      return { ...state, repositorySelectOpen: action.value }
```
Comment (viktor-yakubiv) [inline components/layout/container.jsx]: This should be encapsulated under `CustomSelect`.

**Comment by Joozty** [comment]
I don't consider it important. I just saw it in Figma design and wanted to try to implement it. I do agree it would be great to have some global overlay instead but this would blur all elements. Some of them (like dropdown) show be unblur. Maybe negative `filter: blur(-4px` would work. I don't know. 


Let's not close it. I may find some time to try different approaches. (It's not a priority for me for now.)

**Pull request closed without merging**